WordPress/gutenberg
# Reduce visual prominence of primary actions in table data views, and consolidate primary + secondary actions in ellipsis menu

## What?
1. Reduce the permanent visual prominence of primary actions in table data views.
2. Consolidate primary + secondary actions into the ellipsis menu

## Why?
Permanently displaying the primary actions is quite noisy and adds a lot of repetition to the UI. Revealing them while focussing a specific row can better manage prominence. 

The ellipsis button remains permanently visible so that users can intuit that actions exist at a glance. 

Consolidating all actions into a single menu is helpful for orientation, and means we can use a single reference point for the right-click interaction (https://github.com/WordPress/gutenberg/issues/59608).

## How?
Primary actions are revealed using the same mechanics as the checkbox. That is; when a `tr` has `is-hovered` class, or has `focus-within` the actions appear. Secondary actions (via ellipsis menu) are permanently visible.

## Testing Instructions
* Open a data view such as Pages or Templates
* Select Table layout
* Observe that primary actions appear only when hovering a row, or when focus is within a row.

https://github.com/WordPress/gutenberg/assets/846565/312d8789-e8e9-403b-877e-ef5aaccfa469

**Note**: I'd say this is quite closely related to https://github.com/WordPress/gutenberg/pull/59551. The placement of the Trash action in the menu seems a little odd. This would make more sense:

– Edit
– View
– Revisions
– Trash

Ideally these PR's are reviewed in tandem.
